src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -577,9 +578,9 @@ export class StudyViewPageStore {
     }
 
     @autobind
-    @action updateSelectedGenes(query: SingleGeneQuery[], genesInQuery: Gene[]) {
+    @action updateSelectedGenes(query: SingleGeneQuery[]) {
         this.geneQueries = query;
-        this.queriedGeneSet = new ObservableMap(stringListToSet(genesInQuery.map(gene => gene.hugoGeneSymbol)))
+        this.queriedGeneSet = new ObservableMap(stringListToSet(query.map(gene => gene.gene)))
```

Comment:
again this looks like computation of geneQueries.   by keeping them separate, we have now three sources of truth.

Reply:
Thanks for pointing it out. I actually able to get rid of `queriedGeneSet`.